moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -959,6 +968,54 @@ void ompl_interface::ModelBasedPlanningContext::unregisterTerminationCondition()
   ptc_ = nullptr;
 }
 
+void ompl_interface::ModelBasedPlanningContext::omplPlannerStatusToMoveItErrorCode(
```

Comment:
Why does this use an output parameter and why does it log?

Reply:
I'll get rid of the output parameter.

As far as logging goes, this seems like a pretty good place to do it. We are already enumerating all possible cases, so... that code would have to be duplicated if the logging moved elsewhere. Maybe the function name just needs to be changed to indicate it logs? Like,  `logOmplPlannerStatus()`

Reply:
Maybe do something like this and then the call site can decide what they want:

```cpp
// convert ompl status into moveit error code 
moveit_msgs::msg::MoveItErrorCodes toMoveItError(ompl::base::PlannerStatus status);

// test if ompl status or moveit error code is success 
bool isSuccess(ompl::base::PlannerStatus status);
bool isSuccess(moveit_msgs::msg::MoveItErrorCodes error_code);

// convert ompl status or moveit code into string, used in logging
std::string toString(ompl::base::PlannerStatus status);
std::string toString(moveit_msgs::msg::MoveItErrorCodes error_code);

// log warning if ompl status or moveit code is error
void logWarningIfError(ompl::base::PlannerStatus status);
void logWarningIfError(moveit_msgs::msg::MoveItErrorCodes error_code);
```

At the call site where you need to create a moveit error and log the ompl status you can do this:

```cpp
auto ompl_status = ompl_simple_setup_->getLastPlannerStatus();
logWarningIfError(ompl_status);
result.val = toMoveItError(ompl_status);
```

In model_based_planning_context.cpp it can make the code simpler if you use a `using` statement for success and failure. This is fine to do because it is not in a header and won't propagate into other files.

```cpp
using SUCCESS = moveit_msgs::msg::MoveItErrorCodes::SUCCESS;
using FAILURE = moveit_msgs::msg::MoveItErrorCodes::FAILURE;
```

Then, combined with the isSuccess functions you can make those lines where you need to test if both the moveit error code and the ompl status are good results and return a success/failure moveit code can become a bit less verbose:
```cpp
result.val = (isSuccess(result.val) && isSuccess(ompl_status)) ? SUCCESS : FAILURE;
```

These are just ideas for how you can make some smaller functions you can combine to make the error types easier to use in various places. The important idea here is just separating the function of converting the types from the function that logs.

Reply:
Thank you for getting rid of the output parameter and using a return type. Above are just some suggestions on how we could maybe make this code a bit more flexible and easy to understand at the call site of the function.

Reply:
@AndyZe I'm also happy to accept this change as is and try to make the change I described above myself. Would you rather that?

Reply:
Give me a couple days. I want to try what you've suggested but prob won't get to it today